pkg/plugin/flavor/swarm/flavor.go

```diff
@@ -218,6 +218,28 @@ func (s *baseFlavor) Healthy(flavorProperties *types.Any, inst instance.Descript
 		return flavor.Unknown, nil
 
 	case len(nodes) == 1:
+		if nodes[0].Status.State != swarm.NodeStateReady {
+			log.Warn("Node is not ready",
+				"id", nodes[0].ID,
+				"hostname", nodes[0].Description.Hostname,
+				"state", nodes[0].Status.State)
+			return flavor.Unhealthy, nil
+		}
+		if nodes[0].Spec.Role == swarm.NodeRoleManager {
```

Comment:
Since this function is used for both manager and workers, can we make a `switch role` and check for the cases of `Manager` and `Worker` and return healthy/unhealthy accordingly?  I see only the manager case here..

Reply:
All of the code in this conditional is only applicable to a manager node (since it's checking `ManagerStatus.Reachability`). The "common" check is for the `ready` state (which is above). Do you still want a switch if only the manager role is going to have a case statement?

Reply:
Ok I see your point.  It's saying that the node has to be minimally 'Ready' and for managers another level of checks will be run to determine the health.

Let's keep it as-is then.